freqtrade/commands/data_commands.py

```diff
@@ -64,8 +64,6 @@ def start_download_data(args: Dict[str, Any]) -> None:
     try:
 
         if config.get('download_trades'):
-            if config.get('trading_mode') == 'futures':
-                raise OperationalException("Trade download not supported for futures.")
```

Comment:
Please keep this here (including the test for it).

the ONLY reason to use freqtrade in this mode (download_trades  / `--dl-trades`) is on kraken - which is a far way from supporting futures (ccxt does not support kraken futures at the moment).

Reply:
Sorry, I don't understand the reference to the kraken exchange.... If my trading mode is futures, I'd want to download futures, no? Shouldn't any guards look at _supported_trading_mode_collateral_pairs for the individual specified exchange? I'm probably missing something obvious as I'm still very much learning this codebase.

BTW, I'm trying to get this working with binance and bybit. Bybit is problematic as it doesn't align with the funding ohlcv approach; no fetch_funding_history or 8h timeframes.

Reply:
The area the above "removal" is in (the if-branch) will not run for binance (nor for bybit, as far as i know) - unless you're using a wrong configuration or `--dl-trades` (which you should not use).

The reference about kraken refers to this segment of [the docs](https://www.freqtrade.io/en/stable/data-download/#trades-tick-data) - which definitely does not apply to binance - and should not apply to bybit (i think they do provide historic ohlcv data).

Reply:
Got it, thanks!